src/event/event_kevent.c

```diff
@@ -388,16 +392,18 @@ _dispatch_kevent_print_error(dispatch_kevent_t ke)
 	switch (ke->data) {
 	case 0:
 		return;
+#if DISPATCH_USE_KEVENT_QOS
 	case ERANGE: /* A broken QoS was passed to kevent_id() */
 		DISPATCH_INTERNAL_CRASH(ke->qos, "Invalid kevent priority");
+#endif
 	default:
 		// log the unexpected error
 		_dispatch_bug_kevent_client("kevent", _evfiltstr(ke->filter),
 				!ke->udata ? NULL :
 				ke->flags & EV_DELETE ? "delete" :
 				ke->flags & EV_ADD ? "add" :
 				ke->flags & EV_ENABLE ? "enable" : "monitor",
-				(int)ke->data, ke->ident, ke->udata, du);
+				(int)ke->data, ke->ident, (uint64_t)ke->udata, du);
```

Comment:
Is this explicitly sized to `uint64_t` or would `uintpr_t` be more appropriate?

Reply:
But doesn't that suggest that this is a pointer type (i.e. `uintptr_t`)?

Reply:
Yes, it's a `void *`. The code deliberately passes the pointer to `_dispatch_bug_kevent_client` but the argument is integral (as is `uintptr_t`). Removing the cast gives an error:
```
error: incompatible pointer to integer conversion passing 'void *' to parameter of type 'uint64_t' (aka 'unsigned long long')
```